14-microbiome-diversity.Rmd

```diff
@@ -106,15 +106,33 @@ between communities (samples).
 Technically, beta diversities are usually represented as `dist`
 objects, which contain triangular data describing the distance between
 each pair of samples. These distances can be further subjected to
-ordination with methods such as multi-dimensional scaling (also known
-as PCoA) to retrieve reduced dimensions for further evaluation or
-visualization. [TODO we could here instead link to ordination chapter?]
+ordination. Ordination is a common concept in ecology that aims to
+reduce the dimensionsionality of the data for further evaluation or
+visualization. Ordination techniques aim to capture as much of
+essential information in the data as possible in a lower dimensional
+representation.  Dimension reduction is bound to loose information but
+the common ordination techniques aim to preserve relevant information
+of sample similarities in an optimal way, which is defined in
+different way by different methods. [TODO add references and/or link
+to ordination chapter instead?]
+
+Some of the most common ordination methods in microbiome research
+include Principal Component Analysis (PCA), metric and non-metric
+multi-dimensional scaling (MDS, NMDS), The MDS methods is also known
+as Principal Coordinates Analysis (PCoA). Other recently popular
+techniques include t-SNE and UMAP. 
+
+Some of the above-mentioned ordination methods (PCA, MDS/PCoA) allow
```

Comment:
As far as I know, this only true for SVD/PCA and not for MDS,PCoA although `cmdscale` returns eigenvectors.

Reply:
This is commonly used with MDS/PCoA in microbiome research. 

It does not have an equally straightforward linear interpretation as PCA but it still seems to have [sensible interpretations](https://stats.stackexchange.com/questions/68680/how-to-interpret-variation-explained-by-principal-coordinates) although I must admit that I would still like to have a bit better intuition, too.

Reply:
If we can get a citation or a contribution from a statistician on this, this would be very helpful. The PR to scater I prepared added the support for the return of `eig` and `GOF` for `runMDS`. I refrained from calculating the variance, since I wasn't sure on it. In principle the same interfaces for `runPCA`/`plotPCA` can be used here as well.

(fyi: `runMDS2` will be gone for the release of `mia`)

Reply:
I hope that I should have the required training (statistical ML) to pick up this one. We used to go through all this back in the days. But if someone notices any issues below I will be curious to learn more as this is a quick recap.

PCoA (Classical MDS) aims to minimize so-called "stress" ie. difference between pairwise similarities between the data points in the original (high-dimensional) vs. projected (low-dimensional) space. 

PCA is a special case of PCoA when we operate with Euclidean distances.

With Euclidean distances, we can minimize the stress by projecting to the maximal variance axes. Then the "proportion of explained variance" becomes a rather natural measure for how well the projected distances corresponds to the original distances.

With non-Euclidean distances, the PCoA uses a trick where the pointwise distances (or dissimilarities) are first cast into similarities a Euclidean space (with some information loss ie stress) and then projected to the maximal variance axes. In this case, the maximal variance axes do not any more directly measure the correspondence of the projected distances and original distances. Instead, they measure that correspondence in the already transformed space and interpretation is less straightforward. If the original space is sufficiently Euclidean, this may still be a good approximation (but then why not just use PCA). Explained variance percentages are still sometimes being used with PCoA. Most users probably never thought about it. 

We are probably more interested in measuring the correspondence between the projected and original distances i.e. the stress. Or relative stress, which is better comparable across different data sets: `sum((distEucSC - distMDS)^2)/sum(distEucSC^2)`. This [note](https://www.huber.embl.de/users/klaus/Teaching/statisticalMethods-lab.pdf) from Huber lab gives nice examples. 

Therefore, in fact, I would propose abandoning the idea of showing explained variance percentages for other than PCA ordination.

We could, however, add calculation of the stress and potentially the Shepard plot since these could be rather informative. These could be added either in the package directly, or in OMA. I am not sure what would be the best. Adding to OMA is more light weight at this point perhaps.

Reply:
Ah, I should be able to handle it per training. 

PCA is special case of PCoA with Euclidean distances. Then the variance maximizing procedure and explained percentages reflect directly the level of correspondence between the ordination and original space. With non-Euclidean distances the interpretation is different: the dissimilarities are first cast to Euclidean similarities with a trick, and only then the variance maximizing projections are calculated. This comes with information loss since non-Euclidean dissimilarities cannot, in general, be casted to Euclidean space without information loss. In practice, we are typically interested in quantifying the correspondence between the ordination and original space. We can do that directly by looking at the "stress" function instead. It measures that deviation. See [the note from Huber lab](https://www.huber.embl.de/users/klaus/Teaching/statisticalMethods-lab.pdf) for nice examples.  The explained variances for PCoA would measure how much variance is explained for the data that was already cast in the Euclidean space, and the link with the original data space distances is weaker.

In conclusion, I tend to not recommend showing explained variances for other than PCA. For PCoA, I add an example in OMA beta diversity chapter. We can later consider whether it should be made into its own function or not.